File: parse.js
const fs = require('fs');

function extractText(filePath) {
    let html = fs.readFileSync(filePath, 'utf-8');
    
    html = html.replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, ' ');
    html = html.replace(/<style\b[^<]*(?:(?!<\/style>)<[^<]*)*<\/style>/gi, ' ');
    html = html.replace(/<svg\b[^<]*(?:(?!<\/svg>)<[^<]*)*<\/svg>/gi, ' ');
    html = html.replace(/<[^>]+>/g, '\n');
    
    html = html.replace(/&nbsp;/g, ' ').replace(/&amp;/g, '&');
    html = html.replace(/\n\s*\n/g, '\n').trim();
    
    console.log(html);
}

extractText(process.argv[2]);

File: parse.py
import sys
from bs4 import BeautifulSoup

def extract_text(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        html = f.read()
    soup = BeautifulSoup(html, 'html.parser')
    for script in soup(["script", "style", "meta", "noscript", "svg"]):
        script.extract()
    text = soup.get_text(separator='\n', strip=True)
    print(text)

if __name__ == "__main__":
    extract_text(sys.argv[1])
